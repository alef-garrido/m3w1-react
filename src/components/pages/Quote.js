import React, { useState, useEffect } from 'react';

function Quote() {
  const [quote, setQuote] = useState({
    text: 'loading, please wait...',
    author: '',
  });

  useEffect(() => {
    fetch('https://random-math-quote-api.herokuapp.com/')
      .then((response) => response.json())
      .then((data) => {
        setQuote({
          text: data.quote,
          author: data.author,
        });
      });
  }, []);

  return (
    <div className="quote--container">
      <h3>{quote.text}</h3>
      <p>{quote.author}</p>
    </div>
  );
}

export default Quote;
