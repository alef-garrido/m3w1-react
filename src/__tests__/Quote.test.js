import renderer from 'react-test-renderer';
import Quote from '../components/pages/Quote';

it('renders on screen', () => {
  const tree = renderer
    .create(<Quote />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
