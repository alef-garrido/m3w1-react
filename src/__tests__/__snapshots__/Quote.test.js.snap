// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders on screen 1`] = `
<div
  className="quote--container"
>
  <h3>
    “
    loading, please wait...
    ”
  </h3>
  <p>
    -
    
  </p>
</div>
`;
